apps/site/lib/site_web/controllers/schedule/line.ex

```diff
@@ -39,16 +39,33 @@ defmodule SiteWeb.ScheduleController.Line do
   end
 
   def do_call(
-        %Conn{assigns: %{route: %Route{} = route, direction_id: direction_id}} = conn,
+        %Conn{
+          assigns: %{
+            route: %Route{} = route,
+            direction_id: direction_id
+          }
+        } = conn,
         args
       ) do
+    schedule_direction = conn.query_params["schedule_direction"] || %{}
```

Comment:
@alicia-mbta a somewhat more idiomatic way to write this is via [Map.get/3](https://hexdocs.pm/elixir/Map.html#get/3), something like:

`schedule_direction = Map.get(conn.query_params, "schedule_direction", %{})`

That third argument is the default to use if the key is not found in the map.